solr/core/src/test/org/apache/solr/search/TestBlockCollapse.java

```diff
@@ -44,6 +43,7 @@ public class TestBlockCollapse extends SolrTestCaseJ4 {
   public static void beforeClass() throws Exception {
     // we need DVs on point fields to compute stats & facets
     if (Boolean.getBoolean(NUMERIC_POINTS_SYSPROP)) System.setProperty(NUMERIC_DOCVALUES_SYSPROP,"true");
+    LONG_SHIFT_DIRECTION = random().nextInt(3) - 1; // -1, 0, or 1
```

Comment:
I'm trying to understand why we need randomness here. What's the purpose?

Reply:
We expect the bulk of the functionality we're trying to test to be covered equally regardless of the "shift". So introducing randomness is basically just a way to increase coverage of different variants without paying the cost of running the test more times.

I take your point though; it might be a false economy. Not that much overhead, and as good or better to simply run the test 3 times. In any event that (above) was the intention. (I'm also thinking this is likely a good thing to upstream, so any additional test overhead would affect every build of the upstream project too ...)

Reply:
Gotcha, thanks for the explanation. Fine by me!